README.md

```diff
@@ -1,8 +1,36 @@
 # Delphi Mocks
 
-Delphi Mocks is a simple mocking framework for Delphi XE2 or later. It makes use of RTTI features that are only available in Delphi XE2, however I do hope to be able to get it working with earlier versions of Delphi (2010 or later) at some stage.
+![ Delphi Support ](https://img.shields.io/badge/Delphi%20Support-%20XE2%20...%2010.3%20Rio-blue.svg)
+![ version ](https://img.shields.io/badge/version-%2010.0-a040ff.svg)
 
-# Example
+## Overview
+
+Delphi Mocks is a mocking framework that allows to replace class dependencies with a light implementation of it. This light objects are called test doubles and they are used in unit testing.
+
+Automated testing is mandatory this days and usually technical debt is measured with unit tests coverage. High test coverage allows to apply advanced refactorings and improve even very old code. However in object-oriented world, writing high quality unit test is challenging because of the composition. Object composition is the most popular way of building complex systems. It increases code reusability, but also makes automated testing much more difficult. Why? Because methods of such composed object can not be executed without its dependencies. Delphi Mocks framework enables dynamic substitution of this dependency and provides the opportunity to improve code test coverage.
+
+## Coding to the interfaces
+
+Known also as interface-based programming is a technique for writing interface-based classes and creating composed classes from interfaces - usually uses dependency injection to provide these dependent interfaces to encapsulated private fields of the class.
```

Comment:
> usually uses dependency injection to provide these dependent interfaces to encapsulated private fields of the class.
Now it tools for me like a little to compressed, what maybe makes this sentence to difficult to understand. But fro the other side my idea was to keep overview as much compressed as it is possible. Any suggestions

Reply:
I would avoid going into too much detail here about DI etc. The main purpose of mocking is to do Isolation Testing - ie ensure that you are only testing a very small part of the application in isolation, everything else needed to stand up that part is mocked - take a look at this - https://www.typemock.com/general-unit-testing-page-page3/